protocol.md

```diff
@@ -162,6 +163,67 @@ Receive Firmware Name and Version (after query)
 N  END_SYSEX         (0xF7)
 ```
 
+Report supported Firmata features
+---
+
+Report which optional Firmata features are supported by the firmware and the version of each feature. 
+
+As described in the [Feature registry documentation](https://github.com/firmata/protocol/blob/master/feature-registry.md),
+Firmata features are split into 2 categories: *Core features* and *Optional features*. The feature
+query reports the optional features that are supported by the firmware as well as the version
+of each supported optional feature.
+
+The feature query reduces the need to define a new pin mode for every introduced feature. Going
+forward pin modes will only be defined and maintained for microcontroller specific features such as 
+digital and analog I/O, PWM, I2C, UART, SPI, DAC, etc. The use of pin modes for non microcontroller
```

Comment:
I think we can cover all the considerations mentioned above, if we were to take a step out from the pin level detail and first have a board level description.
Assuming you are not making a tailored solution (in which case any generic Firmata implementation becomes heavy weight and less valuable). As a rule, a person writing remote code doesn't have access to Boards.h, so they don't know anything about the hardware or the board itself. They should be able to query a board for it's feature set (i.e. GPIO, analog, SPI, I2C, etc...). Then, if they are interested in a particular feature like SPI, they could query it specifically. This would allow much greater detail to be returned about a pin or set of pins (i.e. SCLK, SS, MISO and MOSI pins).